responses/11-targets-debug.md

```diff
@@ -6,17 +6,21 @@ If you've written your own functions or scripts before, you may have run into th
 
-In `scipiper`, you can't set a breakpoint in the "normal" way, which would be clicking on the line number after you sourced the script. Instead, you need to use the other method for debugging in R, which requires adding the function call `browser()` to the line where you'd like the function call to stop. 
+In `targets`, you can't set a breakpoint in the "normal" way, which would be clicking on the line number after you sourced the script. Instead, you need to use the other method for debugging in R, which requires adding the function call `browser()` to the line where you'd like the function call to stop and specifying an additional argument when you call `tar_make()`. 
 
 ---
 
-You have a working, albeit brittle, pipeline in your course repository. You can try it out with `scipiper::scmake()`. This pipeline has a number of things you'll work to fix later, but for now, it is a useful reference. The pipeline contains several functions which are defined in `.R` files. 
+You have a working, albeit brittle, pipeline in your course repository. You can try it out with `targets::tar_make()`. This pipeline has a number of things you'll work to fix later, but for now, it is a useful reference. The pipeline contains several functions which are defined in `.R` files. 
 
-So, if you wanted to look at what `download_files` where created within the `download_nwis_data` function, you could set a breakpoint by adding `browser()` to the `"1_fetch/src/get_nwis_data.R"` file:
+So, if you wanted to look at what `download_files` where created within the `download_nwis_data` function, you could set a breakpoint by adding `browser()` to the `"1_fetch/src/get_nwis_data.R"` file. Hint: to quickly navigate to this function source code from your makefile, you can put your cursor on the name of the function then click F2 and it will take you to the correct location in the corresponding source file!
 
 ![browser()](https://user-images.githubusercontent.com/2349007/82158816-bed9bb00-984f-11ea-8892-b2aeb5e4818d.png)
 
-Then, running `scmake()` will land you right in the middle of line 8. Give it a try on your own.
+There is one more step to get your breakpoint to work in `targets`. You will need to add `callr_function = NULL` to your `tar_make()` call. When you run `tar_make(callr_function = NULL)`, you will land right in the middle of line 8. Give it a try on your own.
+
+To navigate while in browser mode, you can use the buttons at the top of your console pane:
+
+![debugnav](https://support.rstudio.com/hc/en-us/article_attachments/201594747/console-toolbar.png)
 
 ---
```

Comment:
Done! Except that I used `data.frame()` because I like to see the type of data structure that it will become in the setup part.

https://github.com/USGS-R/ds-pipelines-targets-2-template/commit/709c508c7865be15fbe7e9583c28c05b25a0a003